src/development/troubleshoot.md

```diff
@@ -161,3 +135,72 @@ where `<project_id>` is the random-character ID of the project.  That can be fou
+2. If you're performing "Sync", "Merge", or "Activate" on an environment and the process is stuck, use [SSH](/development/access-site.md) to connect to the parent environment and identify any long running cron jobs with `ps afx`. Kill the job(s) if you see any.
+3. If you couldn't find any long running process on your environment or parent environment, or you cannot use SSH to connect to your environment, please create a [support ticket](https://platform.sh/support).
+
+
+## Slow or failing build (or deployment)
+
+Build that take long time or fail is a common problem. Most of the time it's related to application issue and they can be hard to troubleshoot without guidance.
+
+Here are a few tips that can help you solve the issues you are experiencing.
+
+### Check for errors in the logs
+
+Invisible errors during the build and deploy phase can cause increase wait times, failed builds and other problems. Investigating each log and fixing errors is essential.
+
+Related documentation: [Accessing logs](https://docs.platform.sh/development/logs.html#accessing-logs)
+
+### Check php.access.log
+
+The php.access.log contains the page executions by php-fpm. It also includes the execution time and peak memory usage of each request. This can be used to find pages and scripts that use to much memory or to much time to finish.
+
+Show 10 slowest page loads in the last 1000 requests: `tail -n 1000 php.access.log | sort -n -k 4 | tail`
+
+### Build and deploy hooks
+
+Hooks are almost always the cause of long build time. If they run into problem they can cause the build to fail or hang indefinitely.
+
+You should test each hook and deploy command in your container (via SSH) with the command `php /app/your_public_folder/your_script.php`.
+
+Furthermore, you can test your hooks with those linux commands to help figure out any problems
+`time $cmd` # Print execution time
+`strace -T $cmd` # Print a system call report
+
+Ps.: Only use these commands in hooks or scripts if you redirect the output to a file by appending ` 2>&1 > strace.log` to your command.
+
+Related documentation: [Build and deploy hooks](https://docs.platform.sh/configuration/app/build.html#hooks)
+
+### Cron jobs
+
+In order to switch containers, we need to wait for all the current running tasks to finish. Long running or failing cron jobs will prevent the container to switch.
+
+First, make sure your custom cron jobs’ execution times are low and that they are running properly. Second, investigate your application cron jobs, as they can invoke other services in unexpected ways, which may increase execution time.
+
+**About Drupal**: `drush core-cron` run insalled module’s cron task. Those can be, for example; evicting invalid cache, updating database records, regenerating assets. Be sure to constantly benchmark the `drush core-cron` command in all your environments, as it is a common source of performance issues.
```

Comment:
This should probably be a note? Grep for "note" in the docs, there's some markup for that iirc.